Svc/FileManager/FileManager.cpp

```diff
@@ -156,10 +164,106 @@ void FileManager ::FileSize_cmdHandler(const FwOpcodeType opCode, const U32 cmdS
     this->sendCommandResponse(opCode, cmdSeq, status);
 }
 
+void FileManager ::ListDirectory_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq, const Fw::CmdStringArg& dirName) {
+    Fw::LogStringArg logStringDirName(dirName.toChar());
+    
+    // Check if we're already listing a directory
+    if (m_listState == LISTING_IN_PROGRESS) {
+        this->log_WARNING_HI_ListDirectoryError(logStringDirName, static_cast<U32>(Os::Directory::OTHER_ERROR));
```

Comment:
Minor: can you pass `dirName` straight in without making a `logStringDirName` variant?  I think that should work now.  Note: this is a new feature.